src/components/Report.tsx

```diff
@@ -21,14 +49,49 @@ const Report: React.FC = () => {
   const [students, setStudents] = useState<Student[]>([]);
   const [gradeScale, setGradeScale] = useState<GradeScale[]>([]);
 
+  // Set the total number of class sessions
+  const totalSessions = TOTAL_SESSIONS;
+
   useEffect(() => {
-    const processedStudents = gradesData.students.map((student) => ({
+    // Process students and attendance data
+    const processedStudents: Student[] = (
+      gradesData.students as RawStudent[]
+    ).map((student) => ({
       ...student,
       overallGrade: student.overallGrade
         ? parseFloat(student.overallGrade)
         : null,
+      courseName:
+        student.courseName || attendanceData.courseName || "Unknown Course",
     }));
-    setStudents(processedStudents);
+
+    const processedAttendance = attendanceData.studentAttendance.map(
+      (student) => ({
+        name: student.name,
+        absences: getAbsences(student.attendance),
+        lates: getLates(student.attendance),
+      })
+    );
+
+    const studentsWithAttendance: Student[] = processedStudents.map(
+      (student) => {
+        const attendanceRecord = processedAttendance.find(
+          (record) =>
+            record.name.toLowerCase() === student.firstName.toLowerCase()
+        );
+        return {
+          ...student,
+          attendance: attendanceRecord
+            ? {
+                absences: attendanceRecord.absences,
+                lates: attendanceRecord.lates,
+              }
+            : undefined,
+        };
+      }
+    );
+
+    setStudents(studentsWithAttendance);
```

Comment:
You might want to replace this and `setGradeScale` with `useMemo` instead and split them.